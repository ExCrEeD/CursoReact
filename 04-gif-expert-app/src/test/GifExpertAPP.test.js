import {shallow} from 'enzyme'
import { GifExpertAPP } from '../GifExpertAPP';

describe('pruebas GifExpertAPP', () => {
    let wrapper = shallow(<GifExpertAPP />);
    test('debe de renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostar una lista de categorias', () => {
        
        const categories = ['one piece','naruto'];
        let wrapper = shallow(<GifExpertAPP defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    

});
