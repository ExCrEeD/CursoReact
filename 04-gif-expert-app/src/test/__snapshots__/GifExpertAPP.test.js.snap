// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifExpertAPP debe de mostar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertAPP
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one piece"
      key="one piece"
    />
    <GifGrid
      category="naruto"
      key="naruto"
    />
  </ol>
</Fragment>
`;

exports[`pruebas GifExpertAPP debe de renderizar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertAPP
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
